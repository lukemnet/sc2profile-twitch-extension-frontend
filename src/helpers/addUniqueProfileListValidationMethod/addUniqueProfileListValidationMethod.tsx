import { YupInstance, AssertingTestFunction } from 'yup';
import isProfileListUnique from 'helpers/isProfileListUnique/isProfileListUnique';
import addValidationMethod from 'helpers/addValidationMethod/addValidationMethod';
import { ValidationMethod } from 'types';

const addUniqueProfileListValidationMethod = (yupInstance: YupInstance) =>
  addValidationMethod({
    validator: yupInstance,
    schemaType: 'array',
    name: 'unique',
    method: function (message: string) {
      return this.test('unique', message, isProfileListUnique as AssertingTestFunction<boolean>);
    } as ValidationMethod,
  });

export default addUniqueProfileListValidationMethod;
