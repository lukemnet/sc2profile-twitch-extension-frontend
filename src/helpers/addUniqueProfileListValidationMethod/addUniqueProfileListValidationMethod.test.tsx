import * as Yup from 'yup';
import addUniqueProfileListValidationMethod from './addUniqueProfileListValidationMethod';

jest.mock('src/helpers/isProfileListUnique/isProfileListUnique', () => {
  return jest.fn().mockImplementation(() => true);
});

it('adds validation method correctly', () => {
  const yupInstance = addUniqueProfileListValidationMethod(Yup);
  expect(yupInstance.array()
    .of(
      yupInstance.string()
    )
    // istanbul ignore next
    .unique('duplicate phone', jest.fn().mockImplementation(() => true)), 
  ).toMatchSnapshot();
  // expect(isProfileListUnique).toHaveBeenCalledTimes(1);
});