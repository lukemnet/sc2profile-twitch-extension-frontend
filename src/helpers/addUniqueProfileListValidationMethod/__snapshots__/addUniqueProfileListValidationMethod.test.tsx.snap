// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`adds validation method correctly 1`] = `
ArraySchema {
  "_blacklist": RefSet {
    "list": Set {},
    "refs": Map {},
  },
  "_conditions": Array [],
  "_deps": Array [],
  "_exclusive": Object {
    "unique": false,
  },
  "_mutate": undefined,
  "_options": Object {
    "abortEarly": true,
    "recursive": true,
  },
  "_subType": StringSchema {
    "_blacklist": RefSet {
      "list": Set {},
      "refs": Map {},
    },
    "_conditions": Array [],
    "_deps": Array [],
    "_exclusive": Object {},
    "_mutate": undefined,
    "_options": Object {
      "abortEarly": true,
      "recursive": true,
    },
    "_type": "string",
    "_typeError": [Function],
    "_whitelist": RefSet {
      "list": Set {},
      "refs": Map {},
    },
    "tests": Array [],
    "transforms": Array [
      [Function],
    ],
  },
  "_type": "array",
  "_typeError": [Function],
  "_whitelist": RefSet {
    "list": Set {},
    "refs": Map {},
  },
  "tests": Array [
    [Function],
  ],
  "transforms": Array [
    [Function],
  ],
}
`;
